libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -138,9 +138,25 @@ void SyntaxChecker::endVisit(ModifierDefinition const& _modifier)
 	m_placeholderFound = false;
 }
 
-bool SyntaxChecker::visit(WhileStatement const&)
+void SyntaxChecker::checkSingleStatementVariableDeclaration(ASTNode const* _statement)
+{
+	auto varDecl = dynamic_cast<VariableDeclarationStatement const*>(_statement);
+	if (varDecl)
+		m_errorReporter.syntaxError(_statement->location(), "Invalid variable declaration. Please declare it inside a block.");
```

Comment:
I'm actually not sure whether this is the best error message, since the declaration will still be in vain, even if inside a block, while this error somewhat suggests that this would make perfect sense (the only reason for a single variable declaration in a block that I would ever see would be to silence a warning about an ignored return value of something like ``call``, while actually wanting to ignore it...)... I don't really have a better suggestion, though, so maybe it's just fine...

"Variable declarations can only be used inside blocks."? A bit more neutral, but maybe not that good either...

Reply:
Changed and added tests.